jvm/ml4ir-inference/src/test/integration/ml4ir/inference/tensorflow/TensorFlowInferenceIT.scala

```diff
@@ -53,11 +89,109 @@ class TensorFlowInferenceIT extends TestData {
     return Source.fromFile(csvFile).getLines.toList.tail.flatMap(extractColumnValues)
   }
 
+  def validateRankingScores(query: StringMapQueryAndPredictions,
+                     sequenceExample: SequenceExample,
+                     scores: Array[Float],
+                     numDocs: Int) = {
+    val docScores = scores.take(numDocs)
+    val maskedScores = scores.drop(numDocs)
+    docScores.foreach(
+      score => assertTrue("all docs should score non-negative", score > 0)
+    )
+    for {
+      maskedScore <- maskedScores
+      docScore <- docScores
+    } {
+      assertTrue(
+        s"docScore ($docScore) should be > masked score ($maskedScore)",
+        docScore > maskedScore
+      )
+    }
+    if (query.predictedScores != null) {
+      assertArrayEquals("scores aren't close enough: ", docScores, query.predictedScores, 1e-2f)
```

Comment:
it looks like this actually passes with 1e-4 (but not 1e-6) which is better than what I was seeing before.

Reply:
1e-6 passes locally, let's see what is the CI saying.